src/pages/ProductDetail/index.jsx

```diff
@@ -24,14 +25,32 @@ function ProductDetail() {
   const { data: product, isLoading } = useSearchProducts({
     productId,
   });
-  const [productCount, setProductCount] = useState(1);
+  const { cart, setProductCountToCart } = useCart();
+  const currentProductInCart = cart.items.filter(
+    (item) => item.id === productId
+  )[0];
 
-  const handleProductCountChange = (e) => {
-    setProductCount(e.target.value);
-  };
+  const [productCount, setProductCount] = useState(() => {
+    let productInCartCount = 1;
+    if (currentProductInCart && currentProductInCart.count > 0) {
+      productInCartCount = currentProductInCart.count;
+    }
+    return productInCartCount;
+  });
 
   const { results: [productItem = {}] = [] } = product;
 
+  const handleProductCountChange = (e) => {
+    const newProductCount = Number(e.target.value);
+    if (newProductCount > productItem.data?.stock) {
```

Comment:
## Feel free to not implement, only a suggestion.
I see this same logic in the `ProductsTable` component, which I understand it's for the condition to update the quantity in the cart depending on the product stock. But, I think you can move this logic to the reducer so from here you only call the function to add products to the cart.